### PR TITLE
docs: Update dns sections for partition query format and virtual IPs

### PR DESCRIPTION
**Pull request by kyhavlov**

### CODE REVIEW

File: website/content/docs/discovery/dns.mdx [hunk 1]
```diff
@@ -335,37 +350,22 @@ using the [`advertise-wan`](/docs/agent/options#_advertise-wan) and
...
-With `prefer_namespace` set to `false` the namespace may be omitted and will be defaulted
-to the `default` namespace:
+Consul Enterprise supports resolving namespaced and partitioned services via DNS.
+To maintain backwards compatibility existing queries can be used and these will
+resolve services within the `default` namespace and partition. However, for resolving
+services from other namespaces or partitions the following form can be used:
 
 ```text
-[tag.]<service>.service.<datacenter>
+[tag.]<service>.service[.<namespace>.ns][.<partition>.ap][.<datacenter>.dc].<domain>
```
Comment (freddygv) [inline website/content/docs/discovery/dns.mdx:361]: Should the brackets be removed from the later parts, since they're not optional the way tags are?

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/529252.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 9576e0f9dd1c3b58a9dc43f1e09c80c244022186 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 9576e0f9dd1c3b58a9dc43f1e09c80c244022186 onto `release/1.11.x` succeeded!

**Pull request merged**